docs/howto/operate/ci-cd/split-repo2docker.md

```diff
@@ -0,0 +1,77 @@
+# Split up an image for use with r2d-action
+
+Sometimes we have user images defined in a repo, and we want to extract
+it to a standalone repo so it can be used with the [repo2docker action](https://github.com/jupyterhub/repo2docker-action).
+But we want to retain full history of the image still, so we can look
+back and see why things are the way they are, as well as credit the people
+who contributed over time. This page documents the git-fu required to
+make this happen.
+
+1. Clone the base repository you are extracting the image from.
+   We will be performing destructive operations on this clone,
+   so it has to be a fresh clone.
+
+    ```bash
+    git clone https://github.com/utoronto-2i2c/jupyterhub-deploy utoronto-image --origin source
+    ```
+
+    We name the directory `utoronto-image` as that would be the name of the
+    repo with just the user image. We ask git to name the remote it creates be
+    called `source`, as we will create a new GitHub repo later that will be `origin`.
+
+2. Make sure [git-filter-repo](https://github.com/newren/git-filter-repo)
```

Comment:
IIRC, I have used `git subtree split` in the past and I believe it already comes with git...
Do you prefer the `git-filter-repo` tool because of some additional features it has?
Just curious 😉 !

Reply:
I've used `git filter-branch` before, and if you try it now, it prints out this official warning:

```
➜ git filter-branch
WARNING: git-filter-branch has a glut of gotchas generating mangled history
         rewrites.  Hit Ctrl-C before proceeding to abort, then use an
         alternative filtering tool such as 'git filter-repo'
         (https://github.com/newren/git-filter-repo/) instead.  See the
         filter-branch manual page for more details; to squelch this warning,
         set FILTER_BRANCH_SQUELCH_WARNING=1.
```

so I decided to take their advice and use filter-repo :D I didn't know subtree split could be used to do this! I've generally never used subtree...